### PR TITLE
systemtests: add upgrade-database-test

### PR DESCRIPTION
**Pull request by pstorz**
This test creates the databases with the create sql statements for Bareos 15
(Database Version 2004) and updates them to the latest version by
running the update_bareos_tables script.

### CODE REVIEW

File: core/scripts/bareos-config-lib.sh.in [hunk 1]
```diff
@@ -762,7 +762,12 @@ get_databases_installed()
 
 get_database_driver_default()
 {
-    DBDRIVER=`get_database_driver`
+    if [ -n "$DBTYPE" ]; then
+      DBDRIVER=$DBTYPE;
+    else
+      DBDRIVER=`get_database_driver`
```
Comment (franku) [inline core/scripts/bareos-config-lib.sh.in]:
```suggestion
      DBDRIVER=$(get_database_driver)
```

File: core/scripts/bareos-config-lib.sh.in [hunk 2]
```diff
@@ -762,7 +762,12 @@ get_databases_installed()
...
+    fi
+
 
     if [ -z "$DBDRIVER" ]; then
         DBDRIVER=`get_databases_installed | head -n 1`
```
Comment (franku) [inline core/scripts/bareos-config-lib.sh.in]:
```suggestion
        DBDRIVER=$(get_databases_installed | head -n 1)
```

File: systemtests/CMakeLists.txt [hunk 3]
```diff
@@ -461,9 +461,7 @@ else() # run systemtests on source and compiled files
 
   get_target_property(FD_PLUGINS_DIR_TO_TEST bpipe-fd BINARY_DIR)
   get_target_property(SD_PLUGINS_DIR_TO_TEST autoxflate-sd BINARY_DIR)
-  if(HAVE_PYTHON)
-    get_target_property(DIR_PLUGINS_DIR_TO_TEST python-dir BINARY_DIR)
-  endif()
+  get_target_property(DIR_PLUGINS_DIR_TO_TEST python-dir BINARY_DIR)
```
Comment (franku) [inline systemtests/CMakeLists.txt]:
```suggestion
if(TARGET python-dir)
  get_target_property(DIR_PLUGINS_DIR_TO_TEST python-dir BINARY_DIR)
endif()
```

File: core/scripts/bareos-config-lib.sh.in (hunk 1 above)
Comment (pstorz) [reply core/scripts/bareos-config-lib.sh.in]: These Backticks are inherited and it does not make sense to only change parts of the script.

**Pull request merged**